stdlib/messaging/rabbitmq/src/main/java/org/ballerinalang/messaging/rabbitmq/RabbitMQUtils.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.messaging.rabbitmq;
+
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.bre.bvm.BLangVMErrors;
+import org.ballerinalang.connector.api.BLangConnectorSPIUtil;
+import org.ballerinalang.connector.api.Struct;
+import org.ballerinalang.model.types.BTypes;
+import org.ballerinalang.model.values.*;
+import org.ballerinalang.util.exceptions.BallerinaException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Util class used to bridge the RabbitMQ connector's native code and the Ballerina API.
+ */
+public class RabbitMQUtils {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RabbitMQUtils.class);
+
+    public static <T> T getNativeObject(Struct struct, String objectId,
+                                        Class<T> objectClass, Context context) {
+        Object messageNativeData = struct.getNativeData(objectId);
+        return verifyNativeObject(context, struct.getName(), objectClass, messageNativeData);
+    }
+
+    public static <T> T getNativeObject(BMap<String, BValue> struct, String objectId,
+                                        Class<T> objectClass, Context context) {
+        Object messageNativeData = struct.getNativeData(objectId);
+        return verifyNativeObject(context, struct.getType().getName(), objectClass, messageNativeData);
+    }
+
+    private static <T> T verifyNativeObject(Context context, String structName, Class<T> objectClass, Object
+            nativeData) {
+        if (!objectClass.isInstance(nativeData)) {
+            throw new BallerinaException(structName + " is not properly initialized.", context);
+        }
+        return objectClass.cast(nativeData);
+    }
+
+    public static void throwRabbitMQException(String message, Context context, Throwable throwable) {
+        LOGGER.error(message, throwable);
+        throw new BallerinaException(message + " " + throwable.getMessage(), throwable, context);
```

Comment:
Since we are throwing the exception, Do we need to log it again?
Since throwable message already captured in throwable object, Do we need to append it to message?